packages/go_router/lib/src/delegate.dart

```diff
@@ -66,14 +67,21 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
         if (didPop) {
           return didPop;
         }
-      } else if (route is ShellRoute) {
-        final bool didPop = await route.navigatorKey.currentState!.maybePop();
+      } else if (route is ShellRouteBase && childRoute != null) {
```

Comment:
can childRoute ever be null in this case? If not we can use assert.

Reply:
or use `late RouteBase childRoute;`

Reply:
Yes, you're right, it should not be possible for a shell route to be the last match in a match list.